packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCaptureDeviceControlling.m

```diff
@@ -0,0 +1,176 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "./include/camera_avfoundation/FLTCaptureDeviceControlling.h"
+
+@interface FLTDefaultCaptureDeviceController ()
+@property(nonatomic, strong) AVCaptureDevice *device;
```

Comment:
Can we directly have `AVCaptureDevice` conform to `FLTDefaultCaptureDeviceControlling` protocol and save this boilerplate in this file?

Reply:
You're right, after playing around a bit I found that it is possible do that. I thought in ObjC that would be problematic and wanted to postpone it to the Swift migration (as Swift protocol extensions give much more flexibility). I had to change some other things like abstract `AVCaptureInput` creation to make this work, but it should be much cleaner now.